haskell/cc.bzl

```diff
@@ -19,19 +19,23 @@ def cc_headers(ctx):
 
   *Internal function - do not use.*
   """
-  hdrs = [
-    hdr
-    # XXX There's gotta be a better way to test the presence of
-    # CcSkylarkApiProvider.
-    for dep in ctx.attr.deps if hasattr(dep, "cc")
-    for hdr in dep.cc.transitive_headers.to_list()
-  ] + [
-    hdr
-    # XXX cc_import doesn't produce a cc field, so we emulate it with
-    # a custom provider.
-    for dep in ctx.attr.deps if CcSkylarkApiProviderHacked in dep
-    for hdr in dep[CcSkylarkApiProviderHacked].transitive_headers.to_list()
-  ]
+  adict = { hdr: None
```

Comment:
I think these are all already depsets, right?  In that case, would it be simpler to use `depset(transitive=...)` to combine them?